src/views/index.js

```diff
@@ -0,0 +1,53 @@
+import React, { useState, useEffect } from "react";
+import { Box } from "@material-ui/core";
+import CharacterBuild from "./CharacterBuild";
+import CharacterClass from "../classes/CharacterClass";
+import EnemyClass from "../classes/EnemyClass";
+import EnemyStatus from "./EnemyStatus";
+import DamageConsole from "./DamageConsole";
+
+export default function DamageCalculator() {
+  const [characterBuild, setCharcterBuild] =
+    useState(
+      new CharacterClass({
+        character_name: "Chongyun",
+        attack_base: 733,
+        attack_total: 1675,
+        DMG_Bonus_Cryo: 46,
+        crit_damage: 194.6,
+      }));
+
+  const [enemyStats, setEnemyStats] = useState(new EnemyClass({}));
+
+  useEffect(() => {
+    console.log(characterBuild);
+    console.log(enemyStats);
+  }, [characterBuild, enemyStats]);
+
+  function updateCharacterField(field, value) {
+    const updatedCharacter = characterBuild;
+    updatedCharacter[field] = value ?? 0;
+    setCharcterBuild(build => build = new CharacterClass(updatedCharacter));
+  }
+
+  function updateEnemyField(field, value) {
+    const updatedEnemyStat = enemyStats;
+    updatedEnemyStat[field] = value ?? 0;
+    setEnemyStats(stats => stats = new EnemyClass(updatedEnemyStat));
```

Comment:
im not sure if making a new char/enemy class on every update slows down performance
i didnt really look into it yet

Reply:
for some reason, setting it to be a normal object will not update it unless the element state changes

Reply:
also just giving normal object doesn't trigger useEffect, so idk